src/core/Spy.js

```diff
@@ -0,0 +1,58 @@
+getJasmineRequireObj().Spy = function (j$) {
+
+  function Spy(name, originalFn) {
+    var args = buildArgs(),
+      /*`eval` is the only option to preserve both this and context:
+        - former is needed to work as expected with methods,
+        - latter is needed to access real spy function and allows to reduce eval'ed code to absolute minimum
+        More explanation here (look at comments): http://www.bennadel.com/blog/1909-javascript-function-constructor-does-not-create-a-closure.htm
+       */
+      /* jshint evil: true */
+      wrapper = eval('(function (' + args + ') { return spy.apply(this, Array.prototype.slice.call(arguments)); })'),
```

Comment:
What if we did:

```js
var spy = /* full spy def., which can close over wrapper */,
    wrapper = (function(spy){ return Function(args, 'return spy.apply(this, arguments);') }(spy))
```

Then you get a "closure" really just formal param from the IIFE) of the spy, and you also get a working `this` value.